src/feasibility_checker.jl

```diff
@@ -168,48 +168,52 @@ function _add_infeasible_nonlinear_constraints(
     return
 end
 
-function _distance_to_set(::Any, set::MOI.AbstractSet)
+function _distance_to_set(::Any, set::MOI.AbstractSet, ::Type)
     return error(
         "Feasibility checker for set type $(typeof(set)) has not been " *
         "implemented yet.",
     )
 end
 
-_distance_to_set(::Missing, ::MOI.AbstractSet) = 0.0
+_distance_to_set(::Missing, ::MOI.AbstractSet, ::Type{T}) where {T} = zero(T)
```

Comment:
So the type argument is only needed because of `missing`? Perhaps we could just return `missing` here and then catch it above.